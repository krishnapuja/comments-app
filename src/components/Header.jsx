import React from "react";
import addCommentIcon from "../add-comment-icon.png";
import notificationIcon from "../notification-icon.png";

const Header = ({ handleAddCommentModalOpen, handleNotificationDisplay, notificationDisplay, notificationFeed }) => (
    <header className="comments-feed-header">
        <h2 className="flex-grow-1">Comments Feed</h2>
        <div className="notification">
            <img className="pad-10" src={notificationIcon} onClick={handleNotificationDisplay}/>
            {notificationFeed.length > 0 && <span className="badge">{notificationFeed.length}</span>}
            <div className="dropdown-content" style={{display: notificationDisplay}}>
                {notificationFeed.map(notifications => <p>{notifications} new messages added</p>)}
            </div>
        </div>

        <img style={{ padding: '10px' }} src={addCommentIcon} onClick={handleAddCommentModalOpen} />
    </header>
);

export default Header;
