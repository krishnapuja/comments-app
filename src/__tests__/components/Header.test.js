import { render } from '@testing-library/react';
import Header from '../../components/Header';

describe('render Header correctly', () => {
  const mockProps = {
    handleAddCommentModalOpen: jest.fn(), 
    handleNotificationDisplay: jest.fn(), 
    notificationDisplay: 'block', 
    notificationFeed: [],
  };

  test('renders the Header snapshot correctly', () => {
    const Apprenderer = render(<Header {...mockProps} />);
    expect(Apprenderer).toMatchSnapshot();
  });
});
