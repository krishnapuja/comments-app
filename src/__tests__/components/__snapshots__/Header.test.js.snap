// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header correctly renders the Header snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="comments-feed-header"
      >
        <h2
          class="flex-grow-1"
        >
          Comments Feed
        </h2>
        <div
          class="notification"
        >
          <img
            class="pad-10"
            src="notification-icon.png"
          />
          <div
            class="dropdown-content"
            style="display: block;"
          />
        </div>
        <img
          src="add-comment-icon.png"
          style="padding: 10px;"
        />
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="comments-feed-header"
    >
      <h2
        class="flex-grow-1"
      >
        Comments Feed
      </h2>
      <div
        class="notification"
      >
        <img
          class="pad-10"
          src="notification-icon.png"
        />
        <div
          class="dropdown-content"
          style="display: block;"
        />
      </div>
      <img
        src="add-comment-icon.png"
        style="padding: 10px;"
      />
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
